### PR TITLE
[WiP][GEARPUMP-368] Restore setting runtime env

### PR DESCRIPTION
**Pull request by karol-brejna-i**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [x] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

**Comment by manuzhang** [comment]
Maybe we'd better change the default runtime environment to `Remote` rather than `Embedded`. It's easy to forget to set it.  

@huafengw may have some comments on this.

**Comment by karol-brejna-i** [comment]
I did the initial version. Will test today.

File: core/src/main/scala/org/apache/gearpump/cluster/main/AppSubmitter.scala [hunk 1]
```diff
@@ -21,12 +21,12 @@ import java.io.File
 import java.net.{URL, URLClassLoader}
 import java.util.jar.JarFile
 
-import org.apache.gearpump.util.{AkkaApp, Constants, LogUtil, Util}
+import org.apache.gearpump.util._
```
Comment (manuzhang) [inline core/src/main/scala/org/apache/gearpump/cluster/main/AppSubmitter.scala]: We'd better not use wildcard imports unless necessaery

**Comment by manuzhang** [comment]
Note that `GearpumpNimbus` also falls into this group

**Comment by karol-brejna-i** [comment]
I did search for AkkaApp usages and there are few groups of them:
1. examples (proper AkkaApp usage)
2. gearpupm-core (comands like submit, kill, etc. plus master and worker)
3. gearpump-services 
3. experiments (storm, yarn, akka-streams)

Some details on experiments (AkkaApp usages):
```
gearpump-experiments-storm  (2 usages found)
    org.apache.gearpump.experiments.storm.main  (2 usages found)
        GearpumpNimbus.scala  (1 usage found)
            47 object GearpumpNimbus extends AkkaApp with ArgumentsParser {
        GearpumpStormClient.scala  (1 usage found)
            27 object GearpumpStormClient extends AkkaApp with ArgumentsParser {
gearpump-experiments-yarn  (3 usages found)
    org.apache.gearpump.experiments.yarn.appmaster  (1 usage found)
        YarnAppMaster.scala  (1 usage found)
            281 object YarnAppMaster extends AkkaApp with ArgumentsParser {
    org.apache.gearpump.experiments.yarn.client  (2 usages found)
        LaunchCluster.scala  (1 usage found)
            160 object LaunchCluster extends AkkaApp with ArgumentsParser {
        ManageCluster.scala  (1 usage found)
            87 object ManageCluster extends AkkaApp with ArgumentsParser {
gearpump-servicesJVM  (1 usage found)
    org.apache.gearpump.services.main  (1 usage found)
        Services.scala  (1 usage found)
            41 object Services extends AkkaApp with ArgumentsParser {
```

This is something I'd like to discuss. Do we change the ancestor here, too?
I need to refresh this yarn code in my head...

**Comment by karol-brejna-i** [comment]
One more thing to discuss:

In yarn code we already have a trait named `Command`:
experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/appmaster/Command.scala


Do we care (maybe we could rename the old one or find a "better" name for the new one) or we don't mind?

**Comment by karol-brejna-i** [comment]
@manuzhang Please, take a look.

**Pull request closed without merging**